### PR TITLE
paginate-get-test

### PR DESCRIPTION
**Pull request by averytribbett**
#60 

Made the GET cards endpoint paginate cards 12 at a time. Also updated the GET cards tests to also make sure the paginate is working properly.

The first test checks the default endpoint which contains no query. When no query is passed the page is defaulted to 1. So the test checks to make sure the endpoint gets the first 12 cards.

The second test checks if a query of "page=2" is passed to the request URL. So the test then tests to make sure it is getting the next 12 cards.

### CODE REVIEW

File: api/src/controllers/cards.controller.ts [hunk 1]
```diff
@@ -24,7 +24,11 @@ export class CardsController {
   // Controller to get all cards this will be used for the main page.
   public getCards = async (req: Request, res: Response, next: NextFunction) => {
     try {
-      const findAllCardsData: PokemonCard[] = await this.card.findAllCards();
+      // Initialize our page to 1 and our limit to 12 (12 cards per page)
+      const page = req.query.page || 1;
```
Comment (Kobe16) [inline api/src/controllers/cards.controller.ts]: Small change: define the page variable as number type here. Since `req.query.page` is a string type, you can convert it to a number type doing [Number()](https://www.tutorialspoint.com/how-to-convert-string-to-number-in-typescript).

File: api/src/controllers/cards.controller.ts [hunk 2]
```diff
@@ -24,7 +24,11 @@ export class CardsController {
...
+      const limit = 12;
```
Comment (Kobe16) [inline api/src/controllers/cards.controller.ts]: Initialize `limit` as a number type too

File: api/src/services/cards.service.ts [hunk 3]
```diff
@@ -8,9 +8,14 @@ import { PokemonCardModel } from '@/models/cards.model';
 
 @Service()
 export class CardService {
-  public async findAllCards(): Promise<PokemonCard[]> {
+  public async findAllCards(page: any, limit: any): Promise<PokemonCard[]> {
```
Comment (Kobe16) [inline api/src/services/cards.service.ts]: Can change types for `page` and `limit` to number types

File: api/src/test/cards.test.ts
Comment (Kobe16) [inline api/src/test/cards.test.ts:1]: Maybe we can also test the edge case where we are on the last page and there are less than 12 cards on the last page. Like, if there are only 8 cards on the last page, test the 8th card there.

File: api/src/controllers/cards.controller.ts [hunk 4]
```diff
@@ -24,7 +24,11 @@ export class CardsController {
...
+
```
Comment (Kobe16) [inline api/src/controllers/cards.controller.ts:40]: May also want to add a small check here to see if the value of `page` is greater than `totalNumberOfCards / limit` (because that page would not exist). If `page` is greater than `totalNumberOfCards / limit`, then we can return an error code or something. 

I'm not sure if there is a way to find out the totalNumberOfCards in our system. If there is not, you can disregard this comment

**Review by Kobe16 — CHANGES_REQUESTED** [review]

File: api/src/controllers/cards.controller.ts (hunk 4 above)
Comment (averytribbett) [reply api/src/controllers/cards.controller.ts:40]: I made changes to the getCards function so the page won't be greater than the number of possible pages. For example, if page 2 is the final page and page 4 is passed the URL the getCards function now automatically just changes the value of the page to the final possible page in this example it would be 2. Not sure if this is best approach but it was what I thought of so the user does not go past the final page.

Comment (Kobe16) [reply api/src/controllers/cards.controller.ts:40]: That is smart, I like that approach. Maybe we should also implement something in the opposite direction too (To check if the inputted page is less than 1 (negative page). Then, we would automatically change the page value to the first page.) I'm going to approve this pull request, but you can add that change whenever in the future

**Review by Kobe16 — APPROVED** [review]
